src/test/kotlin/ar/edu/unahur/obj2/caralibro/UsuarioTest.kt

```diff
@@ -29,6 +49,155 @@ class UsuarioTest : DescribeSpec({
         juana.agregarPublicacion(saludoCumpleanios)
         juana.espacioDePublicaciones().shouldBe(550548)
       }
+
+      it("da me gusta a una publicacion") {
+        val pepe = Usuario()
+        pepe.darMeGusta(videoPlaya)
+        videoPlaya.cantidadDeMeGusta.shouldBe(1)
+        val gonza = Usuario()
+        gonza.darMeGusta(videoPlaya)
+        videoPlaya.cantidadDeMeGusta.shouldBe(2)
+        pepe.darMeGusta(videoPlaya)
+        gonza.darMeGusta(videoPlaya)
+        videoPlaya.cantidadDeMeGusta.shouldBe(2)
+      }
+
+      it("es mas amistoso que otro") {
+        val pepe = Usuario()
+        val gonza = Usuario()
+        val fede = Usuario()
+        pepe.agregarAmigo(fede)
+        pepe.esMasAmistosoQue(gonza).shouldBeTrue()
+      }
+
+      it("puede ver una publicacion"){
+        val pepe = Usuario()
+        val gonza = Usuario()
+        val carlos = Usuario()
+        val fede = Usuario()
+        val maria = Usuario()
+
+        val fotoPlaya = Foto(768, 1024, Publico)
+        gonza.agregarPublicacion(fotoPlaya)
+        pepe.puedeVerPublicacion(fotoPlaya).shouldBeTrue()
+        gonza.puedeVerPublicacion(fotoPlaya).shouldBeTrue()
+
+        val videoOutlast = Video(HD720p, 30, SoloAmigos)
+        gonza.agregarPublicacion(videoOutlast)
+        gonza.agregarAmigo(pepe)
+        pepe.puedeVerPublicacion(videoOutlast).shouldBeTrue()
+        maria.puedeVerPublicacion(videoOutlast).shouldBeFalse()
+        gonza.puedeVerPublicacion(videoOutlast).shouldBeTrue()
+
+        val fotoComida = Foto(768,1024, PrivadoConPermitidos)
+        gonza.agregarPublicacion(fotoComida)
+        gonza.agregarPermitidos(pepe)
+        gonza.agregarPermitidos(maria)
+        pepe.puedeVerPublicacion(fotoComida).shouldBeTrue()
+        maria.puedeVerPublicacion(fotoComida).shouldBeTrue()
+        carlos.puedeVerPublicacion(fotoComida).shouldBeFalse()
+        gonza.puedeVerPublicacion(fotoComida).shouldBeTrue()
+
+        val videoLluvia = Video(HD1080p, 15, PublicoConExcluidos)
+        gonza.agregarPublicacion(videoLluvia)
+        gonza.agregarExcluidos(fede)
+        gonza.agregarExcluidos(maria)
+        pepe.puedeVerPublicacion(videoLluvia).shouldBeTrue()
+        maria.puedeVerPublicacion(videoLluvia).shouldBeFalse()
+        carlos.puedeVerPublicacion(videoLluvia).shouldBeTrue()
+        fede.puedeVerPublicacion(videoLluvia).shouldBeFalse()
+        gonza.puedeVerPublicacion(videoLluvia).shouldBeTrue()
+
+        videoLluvia.cambiarPrivacidad(Publico)
+        pepe.puedeVerPublicacion(videoLluvia).shouldBeTrue()
+        maria.puedeVerPublicacion(videoLluvia).shouldBeTrue()
+        carlos.puedeVerPublicacion(videoLluvia).shouldBeTrue()
+        fede.puedeVerPublicacion(videoLluvia).shouldBeTrue()
+        gonza.puedeVerPublicacion(videoLluvia).shouldBeTrue()
```

Comment:
Idem comment de arriba, este test prueba todos los tipos de privacidad a la vez. Habría que dividir más en `describe`s e `it`s